sdk/java-sdk-spring/src/main/scala/kalix/javasdk/impl/action/ReflectiveActionRouter.scala

```diff
@@ -42,28 +42,24 @@ class ReflectiveActionRouter[A <: Action](
     // lookup ComponentClient
     val componentClients = Reflect.lookupComponentClientFields(action)
 
-    try {
-      componentClients.foreach(_.setCallMetadata(message.metadata()))
-
-      methodInvoker match {
-        case Some(invoker) =>
-          inputTypeUrl match {
-            case ProtobufEmptyTypeUrl =>
-              invoker
-                .invoke(action)
-                .asInstanceOf[Action.Effect[_]]
-            case _ =>
-              invoker
-                .invoke(action, invocationContext)
-                .asInstanceOf[Action.Effect[_]]
-          }
-        case None if ignoreUnknown => ActionEffectImpl.Builder.ignore()
-        case None =>
-          throw new NoSuchElementException(
-            s"Couldn't find any method with input type [$inputTypeUrl] in Action [$action].")
-      }
-    } finally {
-      componentClients.foreach(_.clearCallMetadata())
```

Comment:
We should not clear this. There is one ComponentClient instance per call, so it's safe to keep the same call metadata. 

If someone uses the ComponentClient on a future call back, we risk to start a new span if we clear the metadata. 

In other works, all futures that are spawn from an Action call should and use its ComponentCall instance should use the same call metadata.